### PR TITLE
Pull request for master

### PR DESCRIPTION
**Pull request by sileht**
merge: strict, handle manually closed/merged pr
merge: fixup logging

### CODE REVIEW

File: mergify_engine/actions/merge.py [hunk 1]
```diff
@@ -215,6 +215,15 @@ def update_next_pull(installation_id, installation_token, subscription,
     pull = mergify_pull.MergifyPull.from_number(
         installation_id, installation_token,
         owner, reponame, int(pull_number))
+
+    if pull.g_pull.state == "closed":
+        redis.srem(_get_cache_key(pull), pull.g_pull.number)
+        LOG.debug("pull request have been closed or merged",
```
Comment (jd) [inline mergify_engine/actions/merge.py]: has been

File: mergify_engine/actions/merge.py [hunk 2]
```diff
@@ -249,19 +258,29 @@ def smart_strict_workflow_periodic_task():
...
+            if pull.g_pull.state == "closed":
+                redis.srem(_get_cache_key(pull), pull.g_pull.number)
+                LOG.debug("pull request have been closed or merged",
+                          installation_id=installation_id,
+                          pull_number=pull_number,
+                          repo=owner + "/" + reponame, branch=branch)
+            elif pull.is_behind():
+                # NOTE(sileht): Someone can have merged something manually in
+                # base branch in the meantime, so we have to update it again.
+                LOG.debug("pull request need to be updated again",
```
Comment (jd) [inline mergify_engine/actions/merge.py]: needs

File: mergify_engine/actions/merge.py [hunk 3]
```diff
@@ -249,19 +258,29 @@ def smart_strict_workflow_periodic_task():
...
-
             pull = mergify_pull.MergifyPull.from_number(
                 installation_id, installation_token,
                 owner, reponame, int(pull_number))
-            # NOTE(sileht): Someone can have merged something manually in base
-            # branch in the meantime, so we have to update it again.
-            if not pull.is_behind():
+            if pull.g_pull.state == "closed":
+                redis.srem(_get_cache_key(pull), pull.g_pull.number)
+                LOG.debug("pull request have been closed or merged",
```
Comment (jd) [inline mergify_engine/actions/merge.py]: has

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 2eccd872d000f5b90c19759c5452b0578ea990f0

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 2eccd872d000f5b90c19759c5452b0578ea990f0

**Comment by sileht** [comment]
Pull-request updated, HEAD is now f3d937d923f249ea03e70920adfc9ccd85960423

**Pull request merged**